configure.ac

```diff
@@ -520,21 +520,26 @@ AS_IF([test "x$enable_tx2mon" != xno],[
 AM_CONDITIONAL([HAVE_TX2MON], [test "x$HAVE_TX2MON" = xyes])
 AM_CONDITIONAL([ENABLE_TX2MON], [test "$enable_tx2mon" = "yes"])
 
-if test -z "$ENABLE_KOKKOS_TRUE"; then
-	dnl kokkos needs sos (not the storesos).
-	CHECK_SOS=1
-fi
-if test -z "$ENABLE_DARSHAN_TRUE"; then
-        dnl darshan needs sos (not the storesos).
-        CHECK_SOS=1
-fi
-if test -z "$ENABLE_PROC_STREAMS_TRUE"; then
-        dnl linux_proc_sampler stream stores need sos
-        CHECK_SOS=1
-fi
-if test -z "$ENABLE_STORE_APP_TRUE"; then
-	dnl kokkos needs sos (not the storesos).
-	CHECK_SOS=1
+# List of plugins that require SOS
+required_plugins="store-app proc-streams darshan kokkos"
+
+# Initialize an empty list to collect enabled plugins missing SOS
+missing=""
+
+for plug in $required_plugins; do
+  # Construct the variable name, e.g. ENABLE_DARSHAN_TRUE
+  varname=ENABLE_`echo $plug | tr a-z A-Z | tr '-' '_'`_TRUE
+  # Use eval to get the value of that variable
+  eval val=\$$varname
+
+  if test -z "$val"; then
+    missing="$missing $plug,"
+  fi
+done
+
+# Now check if SOS is enabled
+if test -n "$ENABLE_SOS_TRUE" && test -n "$missing"; then
+  AC_MSG_ERROR([The following features require --enable-sos:$missing. Please re-run configure with --enable-sos.])
```

Comment:
@spwalto, I think the error should be `--with-sos=PATH` is required, please re-run with this configure option.

The --enable-sos is not correct. We should probably remove it.